src/v2i-hub/MapPlugin/src/MapPlugin.h

```diff
@@ -0,0 +1,104 @@
+#pragma once
+
+#ifndef MAPPLUGIN_H_
+#define MAPPLUGIN_H_
+
+#include <atomic>
+#include <iostream>
+#include <map>
+#include <memory>
+#include <mutex>
+#include <sstream>
+#include <stdio.h>
+#include <stdlib.h>
+#include <unistd.h>
+#include <thread>
+#include <time.h>
+#include <sys/time.h>
+#include <string>
+
+#include <tmx/tmx.h>
+#include <tmx/IvpPlugin.h>
+#include <tmx/messages/IvpBattelleDsrc.h>
+#include <tmx/messages/IvpSignalControllerStatus.h>
+#include <tmx/messages/IvpJ2735.h>
+#include <tmx/j2735_messages/J2735MessageFactory.hpp>
+#include "XmlMapParser.h"
+#include "ConvertToJ2735r41.h"
+#include "inputs/isd/ISDToJ2735r41.h"
+
+#define USE_STD_CHRONO
+#include <FrequencyThrottle.h>
+#include <PluginClientClockAware.h>
+
+#include "utils/common.h"
+#include "utils/map.h"
+
+#include <MapSupport.h>
+
+using namespace std;
+using namespace tmx;
+using namespace tmx::messages;
+using namespace tmx::utils;
+
+namespace MapPlugin {
+
+#if SAEJ2735_SPEC < 63
+UPERframe _uperFrameMessage;
+#endif
+
+class MapFile: public tmx::message {
+public:
+	MapFile(): tmx::message() {}
+	virtual ~MapFile() {}
+
+	std_attribute(this->msg, int, Action, -1, );
+	std_attribute(this->msg, std::string, FilePath, "", );
+	std_attribute(this->msg, std::string, InputType, "", );
```

Comment:
Is this "inputType" used? What does this represent?

Reply:
Looks like it was declared but is expected to be empty? I'll find out if we need it or not

Reply:
It's needed to find the file extension type for ISD, TXT, XML, JSON, UPER. Kept it in

Reply:
Cool. The difference between UPER and TXT upload? It seems they both upload bytes but TXT has message frame in it? Why do we need these two different uploads that has similar content? I am not familiar with the usage of this plugin, so I am a bit curious about these different extension types?

Reply:
I think any format we accept should very message content before broadcast. When it is not verified we get exceptions on down stream plugins that attempt to ingest this message. I created the TXT format a while back to avoid dealing with issues with the UPER format logic. These two formats can likely be consolidated since they are both uper hex